File: back/kotlin/app/core/model/BoardType.kt
import kotlinx.serialization.Serializable
import CellType

@Serializable
sealed class BoardType {
    @Serializable
    data class ArrayBoard(val cells: Array<Array<CellType?>>) : BoardType()
    @Serializable
    data class DictionaryBoard(val cells: Map<CellPosition, CellType?>) : BoardType()
}

File: back/go/app/core/model/BoardType.go
package model

type BoardType interface {
	boardMember()
}
type ArrayBoard struct {
Cells [][]*CellType
}

func (*ArrayBoard) boardMember() {}

type DictionaryBoard struct {
Cells map[CellPosition]CellType
}

func (*DictionaryBoard) boardMember() {}
